Source/Scene/Cesium3DTileBatchTable.js

```diff
@@ -236,7 +236,11 @@ Cesium3DTileBatchTable.prototype.applyStyle = function (style) {
       ? style.show.evaluate(feature)
       : DEFAULT_SHOW_VALUE;
     this.setColor(i, color);
-    this.setShow(i, show);
+    if (defined(show)) {
+      this.setShow(i, show);
+    } else {
+      this.setShow(i, false);
+    }
```

Comment:
> Why do you think the fix needs to go in `Cesium3DTileBatchTable.prototype.applyStyle`? Any suggestions on what the fix might look like?

I couldn't figure out a generic way to get it working in `ConditionsExpression.prototype.evaluate` since the return type could be different types. And I think it fits cleaner in `Cesium3DTileBatchTable.prototype.applyStyle` which already has handling for default show and color. What you have is what I had in mind as well. Just a few suggestions...

I think `true` is a better default when `show` is undefined. It matches `DEFAULT_SHOW_VALUE`.

This code can be slightly more concise by using `defaultValue`. Two possible approaches:

```
    var show = defined(style.show) ? style.show.evaluate(feature) : undefined;
    show = defaultValue(show, DEFAULT_SHOW_VALUE);

    ...

    this.setShow(i, show);
```
```
    var show = defined(style.show)
      ? defaultValue(style.show.evaluate(feature), DEFAULT_SHOW_VALUE)
      : DEFAULT_SHOW_VALUE;

    ...

    this.setShow(i, show);
```

`color` could be fixed in this function as well instead of in `ConditionsExpression`. I wasn't super clear in https://github.com/CesiumGS/cesium/pull/9633#discussion_r657900751 but that idea can apply to both show and color.